components/eventForm/EventFormValues.ts

```diff
@@ -0,0 +1,54 @@
+import { HexColorPickerOption } from "../formComponents/HexColorPicker"
+import { EventColor, EventColors } from "../../lib/events/EventColors"
+import { FixedDateRange } from "../../lib/Date"
+import { EditEventInput } from "../../lib/events"
+import { Location } from "../../lib/location"
+
+export type EventFormPlacemarkInfo = {
+  readonly name?: string
+  readonly address?: string
+}
+
+export type EventFormLocationInfo = {
+  readonly coordinates: Location
+  readonly placemarkInfo?: EventFormPlacemarkInfo
+}
+
+/**
+ * Values held by an `EventForm`.
+ */
+export type EventFormValues = {
+  readonly title: string
+  readonly description: string
+  readonly locationInfo?: EventFormLocationInfo
+  readonly dateRange: FixedDateRange
+  readonly color: EventColor
+  readonly shouldHideAfterStartDate: boolean
+  readonly radiusMeters: number
+}
+
+/**
+ * Converts valid `EventFormValues` into an `EditEventInput` type.
+ */
+export const eventEditInputFromFormValues = (values: EventFormValues) => {
+  if (values.title.length === 0 || !values.locationInfo) return undefined
+  return {
+    title: values.title,
+    description: values.description.length > 0 ? values.description : undefined,
+    location: values.locationInfo.coordinates,
+    dateRange: values.dateRange,
+    color: values.color,
+    shouldHideAfterStartDate: values.shouldHideAfterStartDate,
+    radiusMeters: values.radiusMeters
+  } as EditEventInput
+}
+
+export const eventColorOptions = [
+  { color: EventColors.Red, accessibilityLabel: "Red" },
```

Comment:
Better to use a function that transforms the original colors variable so we have one source of truth

Reply:
Ideally yes, but at the same time it would be a little awkward if you consider that it would probably look like this:

```ts
const accessibilityColorMap = {
    // ...
}
export const eventColorOptions = EventColors.all.map((color) => ({ color, accessibilityLabel: accessibilityColorMap[color] ?? "Unknown Color"  }))
```

`"Unknown Color"` would never be returned, so it would seem pretty weird that we should have to handle that case.